frontend/src/v5/ui/routes/userSignup/userSignupForm/userSignupFormStep/userSignupFormStepTermsAndSubmit/userSignupFormStepTermsAndSubmit.component.tsx

```diff
@@ -0,0 +1,251 @@
+	CheckboxContainer,
+	CheckboxMessage,
+	TermsContainer,
+	ErrorContainer,
+	ErrorMessage,
+	FormCheckbox,
+	ErrorIcon,
+	Link,
+	Gap,
+} from './userSignupFormStepTermsAndSubmit.styles';
+
+export interface ITermsAndSubmitFormInput {
+	termsAgreed: boolean;
+	mailListAgreed: boolean;
+	captcha: string;
+	username: string;
+}
+
+type MinimalTermsAndSubmitFormInput = Omit<ITermsAndSubmitFormInput, 'captcha' | 'username'>;
+
+type UserSignupFormStepTermsAndSubmitProps = {
+	updateFields: (fields: any) => void;
+	onSubmitStep: () => void;
+	onComplete: () => void;
+	onUncomplete: () => void;
+	hasUnexpectedError: boolean;
+	fields: ITermsAndSubmitFormInput;
+	isActiveStep: boolean;
+};
+
+export const UserSignupFormStepTermsAndSubmit = ({
+	updateFields,
+	onSubmitStep,
+	onComplete,
+	onUncomplete,
+	hasUnexpectedError,
+	fields,
+	isActiveStep,
+}: UserSignupFormStepTermsAndSubmitProps) => {
+	const DEFAULT_FIELDS: MinimalTermsAndSubmitFormInput = {
+		termsAgreed: false,
+		mailListAgreed: false,
+	};
+
+	const getTermsAndSubmitFields = (): MinimalTermsAndSubmitFormInput => defaults(
+		pick(fields, ['termsAgreed', 'mailListAgreed']),
+		DEFAULT_FIELDS,
+	);
+
+	const {
+		handleSubmit,
+		control,
+		getValues,
+		formState,
+		formState: { isValid: formIsValid },
+	} = useForm<ITermsAndSubmitFormInput>({
+		mode: 'onChange',
+		resolver: yupResolver(UserSignupSchemaTermsAndSubmit),
+		defaultValues: getTermsAndSubmitFields(),
+	});
+
+	const captchaRef = useRef<ReCAPTCHA>();
+	const [submitButtonIsPending, setSubmitButtonIsPending] = useState(false);
+
+	const createAccount: SubmitHandler<ITermsAndSubmitFormInput> = () => {
+		setSubmitButtonIsPending(true);
+		onSubmitStep();
+	};
+
+	const handleCaptchaChange = async (captcha) => {
+		if (!fields.captcha && captcha) {
+			setSubmitButtonIsPending(true);
+			updateFields({ captcha });
+			setSubmitButtonIsPending(false);
+		}
+	};
+
+	useEffect(() => {
+		if (!clientConfigService.captcha_client_key && !fields.captcha) {
+			updateFields({
+				captcha: 'CAPTCHA_IS_DISABLED',
+			});
+		}
+	}, []);
+
+	useEffect(() => {
+		if (formIsValid) {
+			onComplete();
+			if (!fields.captcha) {
+				captchaRef?.current?.execute();
+			}
+		} else {
+			onUncomplete();
+		}
+	}, [formIsValid]);
+
+	useEffect(() => {
+		const formValues = getValues();
+		if (isActiveStep && !isMatch(getTermsAndSubmitFields(), formValues)) {
+			updateFields(formValues);
+		}
+	}, [formState]);
+
+	useEffect(() => setSubmitButtonIsPending(false), [hasUnexpectedError]);
+
+	return (
+		<>
+			<TermsContainer>
+				<CheckboxContainer>
+					<FormCheckbox
+						name="termsAgreed"
+						control={control}
+						label={(
+							<CheckboxMessage>
+								<FormattedMessage
+									id="userSignup.form.terms.termsAndContitions"
+									defaultMessage={`
+										I agree to the {termsAndConditions} and I have
+										read the {privacyPolicy} and the {cookiesPolicy}.
+									`}
+									values={{
```

Comment:
I dont get this one.... :/

Reply:
Since the the `termsAndCondition`, `privacyPolicy`, and `cookiesPolicy` are components (links to be precise), yet they need to be translated, I used this structure. Is it not ok?

Reply:
I think you could go like this, is a bit more clear to me:

```
	<FormattedMessage
		id="userSignup.form.terms.termsAndContitions"
		defaultMessage={`
			I agree to the <termsLink>Terms & Conditions</termsLink> and I have
			read the <privacyLink>Privacy Policy</privacyLink> and the <cookiesLink>Cookies Policy</cookiesLink>.
		`}
		values={{
			termsLink: (label) => (<Link to="/terms" target="_blank">{label}</Link>),
			privacyLink: (label) => (<Link to="/privacy" target="_blank">{label}</Link>),
			cookiesLink: (label) => (<Link to="/cookies" target="_blank">{label}</Link>),
		}}
	/>
```

Like the example in 
https://formatjs.io/docs/react-intl/components/#function-as-the-child